attributes/repo.rb

```diff
@@ -21,6 +21,10 @@
 # ```
 default['ros_buildfarm']['repo']['rsyncd_endpoints'] = Hash[]
 
+# The repository host server name is currently only used to provide
+# a server name when using letsencrypt for https and is environment-specific.
+default['ros_buildfarm']['repo']['server_name'] = nil
```

Comment:
I am bit conflicted about defaulting this to `nil`, the jenkins server
name defaults to `'ros_buildfarm'` and would biff if used with
letsencrypt enabled since it isn't a valid fqdn for the server name
verification.

`nil` breaks the convention of having a default but also gives us the
ability to raise the alarm early if a configuration isn't going to work,
rather than having the acme.sh certificate issue attempt fail.